bundle/artifacts/upload.go

```diff
@@ -41,6 +43,32 @@ func (m *upload) Apply(ctx context.Context, b *bundle.Bundle) error {
 		return fmt.Errorf("artifact source is not configured: %s", m.name)
 	}
 
+	// Check if source paths are absolute, if not, make them absolute
+	for k := range artifact.Files {
+		f := &artifact.Files[k]
+		if !filepath.IsAbs(f.Source) {
+			f.Source = filepath.Join(b.Config.Path, f.Source)
```

Comment:
We could run into more relative path issues if the artifacts are not defined in a config file in the bundle root directory. We can preempt this by getting the path to the directory where this artifact is defined. Similar to how `resources/path` gets its path, it is possible to include a `dyn.Value` field in the `Artifact` struct and the `ToTyped` conversion will include the `dyn.Value` for the artifact in the struct. Then you can use that to figure out the path to the configuration file where it was defined.